

Copyright (C) 2014 Curt Brune <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0

diff --git a/arch/powerpc/kernel/pci-common.c b/arch/powerpc/kernel/pci-common.c
index 12659ab..6ce6a23 100644
--- a/arch/powerpc/kernel/pci-common.c
+++ b/arch/powerpc/kernel/pci-common.c
@@ -1111,6 +1111,13 @@ void __devinit pcibios_setup_bus_devices(struct pci_bus *bus)
 			ppc_md.pci_dma_dev_setup(dev);
 
 		/* Read default IRQs and fixup if necessary */
+		if (pci_read_irq_line(dev) && (dev->devfn != 0)) {
+		     printk(KERN_WARNING
+			    "WARNING: Unable to map IRQ in device tree for pci device "
+			    "%04x:%02x:%02x.%d\n",
+			    pci_domain_nr(dev->bus), dev->bus->number, PCI_SLOT(dev->devfn),
+			    PCI_FUNC(dev->devfn));
+		}
 		pci_read_irq_line(dev);
 		if (ppc_md.pci_irq_fixup)
 			ppc_md.pci_irq_fixup(dev);
